### PR TITLE
Added method to calculate center of mass of Entities

### PR DESCRIPTION
**Pull request by JoaoRodrigues**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Simplified calculation of centers of mass for all entity types. Added a specific method for DisorderedAtoms to keep the logic in Entity simpler.

I wrote a small script to calculate the centers of mass independently for the tests. Possible reviewers: @peterjc, @jgreener64, and @rob-miller.  I'll add an entry to `NEWS.rst` and my name to `CONTRIB.rst` after reviews and CI pass.

### CODE REVIEW

**Comment by jgreener64** [comment]
Looks good to me.

**Comment by JoaoRodrigues** [comment]
Thanks!

File: Bio/PDB/Atom.py [hunk 1]
```diff
@@ -482,6 +483,27 @@ def __repr__(self):
         """Return disordered atom identifier."""
         return "<Disordered Atom %s>" % self.get_id()
 
+    # This is a separate method from Entity.center_of_mass since DisorderedAtoms
+    # will be unpacked by Residue.get_unpacked_list(). Here we allow for a very
+    # specific use case that is much simpler than the general implementation.
+    def center_of_mass(self, geometrical=False):
```
Comment (peterjc) [inline Bio/PDB/Atom.py:489]: Do you have strong views on ``geometical`` versus ``geometric`` here?

**Review by peterjc — APPROVED** [review]
Looks good. Note minor naming question.

I did like the new CodeCov comments on the diff highlighting a couple of lines apparently with no test coverage - are they easily included?

File: Bio/PDB/Atom.py (hunk 1 above)
Comment (JoaoRodrigues) [reply Bio/PDB/Atom.py]: Not at all. Changed it!

Comment (peterjc) [reply Bio/PDB/Atom.py]: Thanks - the shorter works seems clearer to me.

**Comment by JoaoRodrigues** [comment]
>I did like the new CodeCov comments on the diff highlighting a couple of lines apparently with no test coverage - are they easily included?

Thanks! I had a leftover if statement from a previous version of the code. Updated the tests. Also made me realize we don't have a `detach_child` or `delete` method for DisorderedAtoms. You can only add stuff.

**Comment by JoaoRodrigues** [comment]
On second thought, it doesn't make much sense for DisorderedAtom to have a `geometric` keyword. Altlocs are of the same atom, so they have the same element. Removed it and also divided the tests to the `test_PDB_Disordered.py` file.

**Comment by JoaoRodrigues** [comment]
Thanks @peterjc and @jgreener64 for the comments/review.

**Pull request merged**